runner/src/runner.rs

```diff
@@ -252,17 +252,16 @@ impl Runner {
         Ok(pid)
     }
 
-    async fn try_get_release<T: RunnerExt + StorageReader>(runner: &T) -> Result<Option<ClientRelease>, Error> {
-        retry_with_log_async(
-            || {
-                runner
-                    .read_chain_storage::<ClientRelease>(runner.subxt_api())
-                    .into_future()
-                    .boxed()
-            },
-            "Error fetching executable".to_string(),
-        )
-        .await
+    /// A single attempt to the the release
+    async fn try_get_release<T: RunnerExt + StorageReader>(runner: &mut T) -> Result<Option<ClientRelease>, Error> {
+        let api = runner.subxt_api().await?;
+        match runner.read_chain_storage::<ClientRelease>(&api).into_future().await {
+            Ok(x) => Ok(x),
+            Err(err) => {
+                runner.drop_subxt_api(); // we might have lost connection. Rebuild connection next time
```

Comment:
It should be possible to check if the error is related to a lost connection, otherwise this might lead to a lot of bandwidth usage if the chain storage read failed for another reason

Reply:
Agree we should distinguish these errors if possible.

Reply:
> Agree we should distinguish these errors if possible.

Isn't it best to just rebuild the connection anyway, just in case the error matching is too specific? (e.g, error on jsonrpsee rather than on the socket level)